### ISSUE TITLE
interpretation of the sequence divergence distribution 

### ISSUE THREAD

**Issue by sjfleck** [fenderglass/ABruijn#623]
Thank you for creating and maintaining Flye. It's a great tool

I'm exploring my data a little deeper to try and figure out a coverage histogram mystery (one of my species, the haploid assembly mentioned below, is showing half of the expected depth of coverage peak). One of the things i'm trying to completely understand is the "sequence divergence distribution". I'll post the plots below. The first two plots were for flye genome assemblies with high heterozygosity confirmed by CoGe synmap, estimated genome size, duplicated BUSCOs, purge haplotigs, and HapPy:

DEBUG: Sequence divergence distribution: 

    |                   *                      |                                                         
    |                   **                     |                                                         
    |                   ***                    |                                                         
    |                   ***                    |                                                         
    |                  ****                    |                                                         
    |                  *****                   |                                                         
    |                 *******                  |                                                         
    |                ********                  |                                                         
    |                *********                 |                                                         
    |                *********                 |                                                         
    |                *********                 |                                                         
    |               ***********                |                                                         
    |               **************             |                                                         
    |               **************             |                                                         
    |              ***************             |                                                         
    |             *****************   *       **                                                         
    |             ******************* * *     **   *                                                     
    |            ******************************* * **                                                    
    |          *********************************** ****** *    *                                         
    |        ***************************************************** *** *     * *                         
    ----------------------------------------------------------------------------------------------------
    0%        5%        10%       15%       20%       25%       30%       35%       40%       45%       

    Q25 = 0.095, Q50 = 0.11, Q75 = 0.15

DEBUG: Sequence divergence distribution: 

    |                   **                   |                                                           
    |                  ***                   |                                                           
    |                 ****                   |                                                           
    |                 ****                   |                                                           
    |                 *****                  |                                                           
    |                 *****                  |                                                           
    |                 ******                 |                                                           
    |                *******                 |                                                           
    |                *******                 |                                                           
    |                *******                 |                                                           
    |               *********                |                                                           
    |               *********                |                                                           
    |               ***********              |                                                           
    |               ***********              |                                                           
    |               ************             |                                                           
    |              **************            |                                                           
    |             *****************          |                                                           
    |             *****************          |                                                           
    |           ************************ *   |                                                           
    |    *  ***************************************************** ** *  **  *                            
    ----------------------------------------------------------------------------------------------------
    0%        5%        10%       15%       20%       25%       30%       35%       40%       45%       

    Q25 = 0.09, Q50 = 0.1, Q75 = 0.12

The next sequence divergence distribution is for a flye genome assembly with almost no uncollapsed heterozygosity, meaning that it was virtually haploid without needing to use purge haplotigs  (confirmed in the same way as the top two):

DEBUG: Sequence divergence distribution: 

    |                 ***                       |                                                        
    |                 ***                       |                                                        
    |                 ****                      |                                                        
    |                 ****                      |                                                        
    |                 ****                      |                                                        
    |                 ****                      |                                                        
    |                *******                    |                                                        
    |                *******                    |                                                        
    |                *******                    |                                                        
    |                *******                    |                                                        
    |                *********                  |                                                        
    |               **********                  |                                                        
    |               ***********                 |                                                        
    |              **************               |  **                                                    
    |             ***************               |  ****                                                  
    |             ****************             *|******* *                                               
    |             ***************** *       ** ***********                                               
    |            ********************  * ** ***************                                              
    |          ********************************************* **                                          
    |      * ************************************************************   **   *                       
    ----------------------------------------------------------------------------------------------------
    0%        5%        10%       15%       20%       25%       30%       35%       40%       45%       

    Q25 = 0.093, Q50 = 0.11, Q75 = 0.2

What does it mean that the first two had only one peak, but the bottom one has two? Is this at all related to the fact that the first two resulted in highly heterozygous assemblies and the last one was virtually haploid? I searched for information on this, but I wasn't finding anything helpful. Thank you for any insight you're able to provide

**Comment by mikolmogorov** [comment]
Hi @sjfleck 

Just to double check, these are the first sequence divergence plots from each log (Flye produces 3 for each run)? 

If the heterozygosity is within a few %, you'll unlikely to see it from the plots produced from noisy reads. The bump in the 3rd plot may be explained by either extensive repeat content, or some kind of a set of reads with poor quality. Which data type is this?

Best,
Misha

**Comment by sjfleck** [comment]
Misha,
yes, these are only the first sequence divergence plots. I will attach all the plots if that helps (species 1 and 2 have one peak in the first plot and species 3 has 2 peaks in the first plot). Each species was sequenced on its own single ONT MinION flow cell (Guppy version 5.0.11). The error rate of the reads is ~10%. Thank you,
Steve

[Flye_log_select_lines.Species_1.txt](https://github.com/fenderglass/Flye/files/12381021/Flye_log_select_lines.Species_1.txt)
[Flye_log_select_lines.Species_2.txt](https://github.com/fenderglass/Flye/files/12381022/Flye_log_select_lines.Species_2.txt)
[Flye_log_select_lines.Species_3.txt](https://github.com/fenderglass/Flye/files/12381023/Flye_log_select_lines.Species_3.txt)

**Comment by sjfleck** [comment]
I've been exploring these results a little more and species 1, 2, and 3 have estimated coverage of 24, 22, and 22. When the assembly was complete, species 1, 2, and 3 have coverages of 19, 16, and 13, respectively. This is all in the text files I shared above. Since it went from 22x to 13x, does that mean species 3 had the most reads excluded from the assembly process? This initial Flye assembly for species 3 was virtually haploid.

I know that Flye recommends 40x coverage, but after Purge Haplotigs, these assemblies were the expected haploid size and all had ~97% complete BUSCOs and the expected % of duplicated BUSCOs based on an independent reference assembly published shortly after these assembles were finished. Additionally, they have very strong synteny with other published genome assemblies (both before and after using RagTag without mismatch repair or gap filling to scaffold these assemblies using the reference). I feel fairly confident that these assemblies are accurate and useful for my purposes, I'm just trying to understand what occurred during the initial assembly process, especially with species 3. Thanks again,
Steve

EDIT: After looking into the low quality read possibility, species 3's longest reads are pretty low quality. Overall, the mean quality of the reads are the same as the other two, but when I ran Nanostat/nanoplot, the longest reads for species 3 are much lower quality than the other two. Its longest reads have quality scores of 3.6-4.1, while the others are 4.7-12.9 (Species 1) and 5.5-11.8 (Species 2), with most on the higher end. I still stand by the assemblies's quality as mentioned above, but are these low quality reads the reason for the low depth of coverage of the assembly?

**Comment by mikolmogorov** [comment]
@sjfleck could be due to a couple reasons. First, Flye is only using reads with MAPQ > 20 to polish, so maybe you have a lot of reads with low MAPQ (different from read QV). Second, Flye computes median over 100bp windows, and for example if you are using mean, and the coverage is somehow uneven, this may affect the statistics. If you are still unsure, you can map your reads against the assmebly and check the alignments visually in IGV.

**Comment by mikolmogorov** [comment]
Marking as resolved, but feel free to follow up if you have more questions!

**Comment by sjfleck** [comment]
I wanted to add that I solved the discrepancy between the average depth of coverage and the lower coverage peak. It turned out that in one of my assemblies, there were 2 contigs with extremely high depth of coverage. When I ran NCBI BLAST on them, they matched 100% with chloroplast sequences for species I was working with. 

When I compare my 3 genome assemblies, as I progressively removed more of the the high depth of coverage contigs, the average depth of coverage for two of them stayed ~22x and the one assembly with the 2 chloroplast contigs progressively dropped to ~13x, as Flye said in the log file. 

this was a good lesson is making sure plastid sequences are not present in the assembly
